packages/flutter_tools/lib/src/build_info.dart

```diff
@@ -872,8 +872,12 @@ String getLinuxBuildDirectory([TargetPlatform? targetPlatform]) {
 }
 
 /// Returns the Windows build output directory.
-String getWindowsBuildDirectory() {
-  return globals.fs.path.join(getBuildDirectory(), 'windows');
+String getWindowsBuildDirectory([TargetPlatform? targetPlatform]) {
+  final String arch = (targetPlatform == null) ?
+      _getCurrentHostPlatformArchName() :
```

Comment:
Should we use `x64` instead of the host platform's architecture since for now we'll build an x64 binary on Windows Arm64?

Reply:
For now, windows-arm64 does not exist on Flutter, so it's x64 that will be chosen:
- https://github.com/flutter/flutter/blob/58332e95d66c784220ddb3a5d397b598c3aa03cc/packages/flutter_tools/lib/src/build_info.dart#L807

For now, I would prefer to leave it like this, else it's needed to add this target parameter everywhere...

If this does not satisfy you, I can simply hardcode 'x64'. My goal is to keep this PR simple and focused on a simple change (not to introduce the possibility of a new potential targetPlatform).

Reply:
> For now, windows-arm64 does not exist on Flutter, so it's x64 that will be chosen:

Ah gotcha, I didn't notice that.

> I can simply hardcode 'x64'. My goal is to keep this PR simple and focused on a simple change (not to introduce the possibility of a new potential targetPlatform).

Yup, hardcoding seems perfectly reasonable to me.

It looks like `getWindowsBuildDirectory` is only used in two places:

https://github.com/flutter/flutter/blob/9cc4f94397f1f4b4d49ff38a02e5a0708dc9d68e/packages/flutter_tools/lib/src/windows/build_windows.dart#L82

https://github.com/flutter/flutter/blob/9cc4f94397f1f4b4d49ff38a02e5a0708dc9d68e/packages/flutter_tools/lib/src/windows/application_package.dart#L114

I'd lean towards making the argument required and updating those two call sites to hard code `TargetPlatform.windows_x64`. That makes it a bit clearer what's happening from the call sites.

Reply:
Will do it 👍